src/store/store.js

```diff
@@ -355,15 +355,20 @@ var Store = (Ember.Service || Ember.Object).extend({
 			recordRequestCache.savePendingRequest(typeKey, query, promise);
 		}
 
-		return PromiseArray.create({
-			promise: promise.then((payload) => {
-				var records = payload.meta.matchedRecords;
-				this.pushPayload(payload);
-
-				return records.map((record) => {
-					return this.getRecord(record.type, record.id);
-				});
-			})
+		return promise.then((payload) => {
+			return {
```

Comment:
There's also the [`Ember.meta` function](https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-metal/lib/meta.js#L557), which will probably work. I can't remember _exactly_ how it works, but I know [I've used it before.](https://github.com/gordonkristan/ember-computed-indirect/blob/master/addon/utils/indirect.js#L14) 😄